teps/0001-tekton-enhancement-proposal-process.md

```diff
@@ -0,0 +1,471 @@
+a change that impacts other `tektoncd` projects or users cannot be
+merged if there is no `TEP` associated with it. Bug fixes and small
+changes like refactoring that do not affect the APIs (CRDs, REST APIs)
+are not concerned by this. Fixing the behaviour of a malfunctioning
+part of the project does not require a TEP.
+
+This TEP process is related to
+- the generation of an architectural roadmap
+- the fact that the what constitutes a feature is still undefined
+- issue management
+- the difference between an accepted design and a proposal
+- the organization of design proposals
+
+This proposal attempts to place these concerns within a general
+framework.
+
+## Motivation
+
+For cross project or new project proposal, an abstraction beyond a
+single GitHub issue seems to be required in order to understand and
+communicate upcoming changes to the Tekton community.
+
+In a blog post describing the [road to Go 2][], Russ Cox explains
+
+> that it is difficult but essential to describe the significance of a
+> problem in a way that someone working in a different environment can
+> understand
+
+As a project, it is vital to be able to track the chain of custody for
+a proposed enhancement from conception through implementation.
+
+Without a standardized mechanism for describing important
+enhancements, our talented technical writers and product managers
+struggle to weave a coherent narrative explaining why a particular
+release is important. Additionally, for critical infrastructure such
+as Tekton, adopters need a forward looking road map in order to plan
+their adoption strategy.
+
+Before this proposal, there is not a standard way or template to
+create project enhancements, but
+[suggestions](https://github.com/tektoncd/community/blob/master/process.md#proposing-features)
+on proposing a feature. We rely on documents hosted on Google docs,
+without a standard template explaining the change. Once a proposal is
+done, via a design docs, it tends to be hard to follow what happens
+with the proposal: updates on the proposal in reaction to comments,
+state of the proposal (when is it accepted, or rejected).
+
+The purpose of the TEP process is to reduce the amount of "tribal
+knowledge" in our community. This is done by putting in place a gate
+(submitting and getting a TEP merged) that mark a decision after
+having discussed the subject during video calls, on mailing list and
+other means. This process aims to enhance communication and
+discoverability. The TEP process is intended to create high quality
+uniform design and implementation documents for WGs to deliberate.
+
+A TEP is broken into sections which can be merged into source control
+incrementally in order to support an iterative development process. A
+number of section are required for a TEP to get merged in the
+`proposed` state (see the different states in the [TEP
+Metadata](#tep-metadata)). The rest of the section can be updated once
```

Comment:
This last sentence is a bit awkward. Maybe something like "The other sections can be updated after further discussions and agreement from the Working Groups."